pkg/data/account.go

```diff
@@ -0,0 +1,172 @@
+package data
+
+import (
+	gErrors "errors"
+	"fmt"
+	"github.com/Optum/dce/pkg/errors"
+	"github.com/Optum/dce/pkg/model"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/dynamodb"
+	"github.com/aws/aws-sdk-go/service/dynamodb/dynamodbattribute"
+	"github.com/aws/aws-sdk-go/service/dynamodb/dynamodbiface"
+	"github.com/aws/aws-sdk-go/service/dynamodb/expression"
+)
+
+// Account - Data Layer Struct
+type Account struct {
+	AwsDynamoDB    dynamodbiface.DynamoDBAPI
+	TableName      string `env:"ACCOUNT_DB"`
+	ConsistentRead bool   `env:"USE_CONSISTENT_READS" envDefault:"false"`
+}
+
+// Update the Account record in DynamoDB
+func (a *Account) Update(account *model.Account, lastModifiedOn *int64) error {
+
+	var expr expression.Expression
+	var err error
+	var returnValue string
+	// lastModifiedOn is 0 on a create
+	if lastModifiedOn != nil {
+		modExpr := expression.Name("LastModifiedOn").Equal(expression.Value(lastModifiedOn))
+		expr, err = expression.NewBuilder().WithCondition(modExpr).Build()
+		returnValue = "NONE"
+	} else {
+		modExpr := expression.Name("LastModifiedOn").AttributeNotExists()
+		expr, err = expression.NewBuilder().WithCondition(modExpr).Build()
+		returnValue = "NONE"
+	}
+
+	putMap, _ := dynamodbattribute.Marshal(account)
+	_, err = a.AwsDynamoDB.PutItem(
+		&dynamodb.PutItemInput{
+			// Query in Lease Table
+			TableName: aws.String(a.TableName),
+			// Find Account for the requested accountId
+			Item: putMap.M,
+			// Condition Expression
+			ConditionExpression:       expr.Condition(),
+			ExpressionAttributeNames:  expr.Names(),
+			ExpressionAttributeValues: expr.Values(),
+			// Return the updated record
+			ReturnValues: aws.String(returnValue),
+		},
+	)
+	var awsErr awserr.Error
+	if gErrors.As(err, &awsErr) {
+		if awsErr.Code() == "ConditionalCheckFailedException" {
+			return fmt.Errorf(
+				"unable to update account %v no account exists with "+
+					"LastModifiedOn=\"%v\": %w",
+				account.ID, *account.LastModifiedOn, errors.ErrConflict)
+		}
```

Comment:
There's an unhandled error here

Reply:
how isn't it handled?  I might be missing something here.